apps/molgenis-components/src/components/tables/AggregateTable.vue

```diff
@@ -187,31 +190,47 @@ export default defineComponent({
         this.rows.push(row);
       }
     },
-  },
-  created() {
-    if (this.allColumns.length > 0) {
-      this.refColumns = getRefTypeColumns(this.allColumns as IColumn[]);
+    initialize() {
+      if (this.allColumns.length > 0) {
+        this.refColumns = getRefTypeColumns(
+          this.allColumns as IColumn[],
+          this.canView
+        );
+      }
       if (this.refColumns?.length > 0) {
         this.selectedColumn = this.refColumns[0];
         this.selectedRow = this.refColumns[1] || this.refColumns[0];
+        this.fetchData();
       }
-    }
-    this.fetchData();
+    },
+  },
+  watch: {
+    selectedColumn() {
+      this.fetchData();
+    },
+    selectedRow() {
+      this.fetchData();
+    },
```

Comment:
I would prefer to see some handler functions on the @update:modelValue instead of watches, as watches tend to create unwanted side-effects as code changes.

Reply:
I initially did the checks on @update:modelValue but what seems to happen is that that also fires on creation/re-rendering. This resulted in endless loop of fetchData.

Can you try if that also happens on your machine?